llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -41293,6 +41293,154 @@ static SDValue combineShuffle(SDNode *N, SelectionDAG &DAG,
   return SDValue();
 }
 
+// Simplify a decomposed (sext (setcc)). Assumes prior check that
+// bitwidth(sext)==bitwidth(setcc operands).
```

Comment:
I suppose this could technically be moved to TargetLowering.cpp so any ZeroOrNegativeOneBooleanContent target could use it?

Reply:
Ill port

Reply:
So there where no test changes outside of X86.

I would prefer to keep in x86 for now, im a bit worried about whether the float semantics will translate to other targets.
Ill leave a note in simplfyDemandedBits that if we want to extend `ISD::SETCC` it may be worth porting.